smac/utils/sklearn_wrapper.py

```diff
@@ -0,0 +1,324 @@
+            the folds, and the loss minimized is the total loss per sample,
+            and not the mean loss across the folds.
+
+        refit : bool
+            Undocumented function of Scikit-learn
+
+        cv : int, cross-validation generator or an iterable, optional
+            Determines the cross-validation splitting strategy.
+            Possible inputs for cv are:
+              - None, to use the default 3-fold cross validation,
+              - integer, to specify the number of folds in a `(Stratified)KFold`,
+              - An object to be used as a cross-validation generator.
+              - An iterable yielding train, test splits.
+
+            For integer/None inputs, if the estimator is a classifier and ``y`` is
+            either binary or multiclass, :class:`StratifiedKFold` is used. In all
+            other cases, :class:`KFold` is used.
+
+            Refer :ref:`User Guide <cross_validation>` for the various
+            cross-validation strategies that can be used here.
+
+        refit : boolean, default=True
+            Refit the best estimator with the entire dataset.
+            If "False", it is impossible to make predictions using
+            this RandomizedSearchCV instance after fitting.
+
+        verbose : integer
+            Controls the verbosity: the higher, the more messages.
+
+        random_state : int or RandomState
+            Pseudo random number generator state used for random uniform sampling
+            from lists of possible values instead of scipy.stats distributions.
+
+        error_score : 'raise' (default) or numeric
+            Value to assign to the score if an error occurs in estimator fitting.
+            If set to 'raise', the error is raised. If a numeric value is given,
+            FitFailedWarning is raised. This parameter does not affect the refit
+            step, which will always raise the error.
+
+        return_train_score : boolean, default=True
+            If ``'False'``, the ``cv_results_`` attribute will not include training
+            scores.
+            """
+
+        # Current implementation does not like distributions yet.
+        for param, values in param_distributions.items():
+            if not isinstance(values, list):
+                raise ValueError('Not implemented: Wrapper does not work with distributions yet. Please use list. ')
+
+        self.random_state = random_state
+        self.param_distributions = param_distributions
+        self.n_iter = n_iter
+
+        # sets important hyperparameters of search module
+        super(ModelBasedOptimization, self).__init__(
+              estimator=estimator, scoring=scoring, fit_params=fit_params,
+              n_jobs=1, iid=iid, refit=refit, cv=cv, verbose=verbose,
+              pre_dispatch=False, error_score=error_score,
+              return_train_score=return_train_score)
+
+    def fit(self, X, y=None, groups=None):
```

Comment:
in our smac repo, we are using type hints as introduced in python 3.5, see
https://www.python.org/dev/peps/pep-0484/